hyp3_metadata/__main__.py

```diff
@@ -0,0 +1,207 @@
+        processor_version=processor_version,
+    )
+    files = []
+    files.extend(create_product_xmls(payload))
+    files.append(create_readme(payload))
+    files.append(create_dem_xml(payload))
+    files.append(create_browse_xml(payload))
+    files.append(create_inc_map_xml(payload))
+    files.append(create_ls_map_xml(payload))
+    return files
+
+
+def get_dem_resolution(dem_name: str) -> str:
+    data = {
+        'NED13': '1/3 arc seconds (about 10 meters)',
+        'NED1': '1 arc second (about 30 meters)',
+        'NED2': '2 arc seconds (about 60 meters)',
+        'SRTMGL1': '1 arc second (about 30 meters)',
+        'SRTMGL3': '3 arc seconds (about 90 meters)',
+    }
+    return data[dem_name]
+
+
+def get_dem_template_id(dem_name: str) -> str:
+    if dem_name.startswith('EU'):
+        return 'eu'
+    if dem_name.startswith('GIMP'):
+        return 'gimp'
+    if dem_name.startswith('NED'):
+        return 'ned'
+    if dem_name.startswith('REMA'):
+        return 'rema'
+    if dem_name.startswith('SRTM'):
+        return 'srtm'
+    raise NotImplementedError(f'Unkown DEM: {dem_name}')
+
+
+def get_projection(srs_wkt) -> str:
+    srs = osr.SpatialReference()
+    srs.ImportFromWkt(srs_wkt)
+    return srs.GetAttrValue('projcs')
+
+
+def get_granule_type(granule_name) -> Tuple[str, str]:
+    granule_type = granule_name[7:10]
+    if granule_type == 'SLC':
+        return 'SLC', 'Single-Look Complex'
+    if granule_type == 'GRD':
+        return 'GRD', 'Ground Range Detected'
+
+
+# FIXME: in hyp3_rtc_gamma.rtc_sentinel as well -- move to hyp3lib?
+def get_polarizations(product_polarization):
+    mapping = {
+        'SH': ('HH',),
+        'SV': ('VV',),
+        'DH': ('HH', 'HV'),
+        'DV': ('VV', 'VH'),
+    }
+    return mapping[product_polarization]
+
+
+def decode_product(product_dir: Path) -> dict:
+    product_parts = product_dir.name.split('_')
+    user_options = product_parts[-2]
+
+    return {'resolution': int(product_parts[-4][-2:]),
+            'radiometry': 'gamma-0' if user_options[0] == 'g' else 'sigma-0',
+            'scale': 'power' if user_options[1] == 'p' else 'amplitude',
+            'masked': False if user_options[2] == 'u' else True,
+            'filter_applied': False if user_options[3] == 'n' else True,
+            'clipped': False if user_options[4] == 'e' else True,
+            'matching': False if user_options[5] == 'd' else True,
+            'polarizations': product_parts[-5][:2],
+            }
+
+
+def marshal_metadata(product_dir: Path, granule_name: str, dem_name: str, processing_date: datetime, looks: int,
+                     plugin_name: str, plugin_version: str, processor_name: str, processor_version: str) -> dict:
+    payload = locals()
+    payload['metadata_version'] = __version__
+
+    payload.update(decode_product(product_dir))
+
+    payload['granule_type'], payload['granule_description'] = get_granule_type(granule_name)
```

Comment:
might have `get_granule_type` return a `dict` and do `payload.update` the same as for `decode_product`